scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -69,34 +69,53 @@ class EscapeHatch(
   def filter(
       patchesByName: Map[RuleName, Patch],
       ctx: RuleCtx,
-      index: SemanticdbIndex): (Patch, List[LintMessage]) = {
+      index: SemanticdbIndex,
+      isEnabledByGitDiff: Option[Position => Boolean])
+    : (Patch, List[LintMessage]) = {
     val usedEscapes = mutable.Set.empty[EscapeOffset]
     val lintMessages = List.newBuilder[LintMessage]
 
+    def disabledByGitDiff(position: Position): Boolean = {
+      isEnabledByGitDiff match {
+        // --diff is provided, check if in aditions or modifications
+        case Some(fun) => !fun(position)
+        // --diff not provided, it's enabled
+        case None => false
+      }
+    }
+
+    def disabledByEscape(name: String, start: Int): Boolean = {
+      // check if part of on/off/ok blocks
+      val (isPatchEnabled, culprit) = isEnabled(name, start)
+      // to track unused on/off/ok
+      culprit.foreach(escape => usedEscapes += escape.offset)
+      !isPatchEnabled
+    }
+
     def loop(name: RuleName, patch: Patch): Patch = patch match {
       case AtomicPatch(underlying) =>
-        val hasEscape = {
+        val hasDisabledPatch = {
           val patches = Patch.treePatchApply(underlying)(ctx, index)
           patches.exists { tp =>
-            val (isPatchEnabled, culprit) =
-              isEnabled(name.toString, tp.tok.pos.start)
-            culprit.foreach(escape => usedEscapes += escape.offset)
-            !isPatchEnabled
+            disabledByGitDiff(tp.tok.pos) ||
```

Comment:
What happens about "unused // scalafix:ok" warnings with this change?

Reply:
yup, let's evaluate everything and not use a shortcut operator.